source/extensions/filters/http/grpc_http1_reverse_bridge/filter.cc

```diff
@@ -179,6 +179,22 @@ Http::FilterHeadersStatus Filter::encodeHeaders(Http::ResponseHeaderMap& headers
     // We can only insert trailers at the end of data, so keep track of this value
     // until then.
     grpc_status_ = grpcStatusFromHeaders(headers);
+
+    // This is a header-only response, and we should prepend the gRPC frame
+    // header directly.
+    if (end_stream && withhold_grpc_frames_) {
+      Envoy::Buffer::OwnedImpl data;
+      buildGrpcFrameHeader(data, 0);
+      encoder_callbacks_->addEncodedData(data, false);
+
+      // This call exists to ensure that the content-length is set correctly,
+      // regardless of whether we are withholding grpc frames above.
+      headers.setContentLength(Grpc::GRPC_FRAME_HEADER_SIZE);
+
+      // Insert grpc-status trailers to communicate the error code.
+      auto& trailers = encoder_callbacks_->addEncodedTrailers();
```

Comment:
is the correct thing to do here to have [headers][trailers] or edit the headers?  cc @lizan

Reply:
@alyssawilk For this particular filter, the correct response based on the gRPC protocol is actually [headers][grpc_frame_header][trailers] so I think the trailers are needed.